src/main/java/de/qabel/desktop/ui/contact/menu/ContactMenuController.java

```diff
@@ -0,0 +1,174 @@
+
+    @Inject
+    private ClientConfig clientConfiguration;
+
+    @Inject
+    Pane layoutWindow;
+
+    @Inject
+    private TransactionManager transactionManager;
+
+    @FXML
+    AnchorPane menuContact;
+
+    @FXML
+    VBox vboxMenu;
+
+    @FXML
+    Button importButton;
+
+    @FXML
+    Button exportToFileButton;
+
+    @FXML
+    Button importFromFile;
+    @FXML
+    Button importFromQR;
+    @FXML
+    Button searchButton;
+    @FXML
+    Button enterContact;
+    @FXML
+    Button exportContactsToFile;
+    @FXML
+    Button exportContactsToQR;
+
+    Identity identity;
+
+    ImageView imageView;
+
+    public PopOver popOver;
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+        resourceBundle = resources;
+        identity = clientConfiguration.getSelectedIdentity();
+    }
+
+    public void showMenu(double coordX, double coordY) {
+        initializePopOver();
+        createButtonGraphics();
+        popOver.show(menuContact, coordX, coordY);
+    }
+
+    private void createButtonGraphics() {
+        Tooltip.install(importButton, new Tooltip(resourceBundle.getString("contactImport")));
+        Tooltip.install(exportToFileButton, new Tooltip(resourceBundle.getString("contactExport")));
+        importFromFile.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/import_black.png"))));
+        importFromQR.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/qrcode.png"))));
+        searchButton.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/cloud_search.png"))));
+        enterContact.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/pencil.png"))));
+        exportContactsToFile.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/export.png"))));
+        exportContactsToQR.setGraphic(getImage(new Image(getClass().getResourceAsStream("/img/qrcode.png"))));
+    }
+
+    private ImageView getImage(Image image) {
+        imageView = new ImageView();
+        imageView.setImage(image);
+        return imageView;
+    }
+
+    private void initializePopOver() {
+        popOver = new PopOver();
+        popOver.setArrowLocation(PopOver.ArrowLocation.TOP_RIGHT);
+        popOver.setContentNode(new VBox(vboxMenu));
+        popOver.setAutoFix(true);
+        popOver.setAutoHide(true);
+        popOver.setHideOnEscape(true);
+        popOver.setDetachable(false);
+        popOver.setOnHiding(event -> layoutWindow.getChildren().remove(menuContact));
+    }
+
+    public void handleExportContactsButtonAction() throws EntityNotFoundException, IOException, JSONException {
+        tryOrAlert(() -> {
+            try {
+                FileChooser chooser = new FileChooser();
+                chooser.setTitle(resourceBundle.getString("contactDownload"));
+                chooser.setInitialFileName("Contacts.qco");
+                File file = chooser.showSaveDialog(menuContact.getScene().getWindow());
```

Comment:
if you only exportContacts if `file != null` you don't have to catch strange nullpointers